morpheus/_lib/src/messages/control.cpp

```diff
@@ -34,18 +34,146 @@
 namespace py = pybind11;
 using namespace py::literals;
 
+namespace {
+py::object cast_from_json(const morpheus::json_t& source)
+{
+    if (source.is_null())
+    {
+        return py::none();
+    }
+    if (source.is_array())
+    {
+        py::list list_;
+        for (const auto& element : source)
+        {
+            list_.append(cast_from_json(element));
+        }
+        return std::move(list_);
+    }
+    if (source.is_boolean())
+    {
+        return py::bool_(source.get<bool>());
+    }
+    if (source.is_number_float())
+    {
+        return py::float_(source.get<double>());
+    }
+    if (source.is_number_integer())
+    {
+        return py::int_(source.get<morpheus::json_t::number_integer_t>());
+    }
+    if (source.is_number_unsigned())
+    {
+        return py::int_(source.get<morpheus::json_t::number_unsigned_t>());
+    }
+    if (source.is_object())
+    {
+        py::dict dict;
+        for (const auto& it : source.items())
+        {
+            dict[py::str(it.key())] = cast_from_json(it.value());
+        }
+
+        return std::move(dict);
+    }
+    if (source.is_string())
+    {
+        return py::str(source.get<std::string>());
+    }
+    if (source.is_binary())
+    {
+        return source.get_binary().get_py_obj();
+    }
```

Comment:
To be safe, we should utilize the type ID property of the binary objects to ensure that our binary type is definitely a python one. For example,

```cpp
nlohmann::json j = {{"key", "value"}};

j["test"] = nlohmann::json::binary(morpheus::PythonByteContainer(source), typeid(py::object));

if (j["test"].is_binary())
{
    if (j["test"].get_binary().has_subtype() && j["test"].get_binary().subtype() == typeid(py::object))
    {
        return j["test"].get_binary().get_py_obj();
    }

    // Otherwise, throw an error
}
```